docs/fxa-engineering/fxa-sub-platform.md

```diff
@@ -169,13 +169,108 @@ server:
+
+### PayPal Accounts
+
+For development, you will need a PayPal developer account.  Once you are in the PayPal developer dashboard, navigate to "Accounts" under the Sandbox section of the menu.  Here you should create a pair of personal and business sandbox accounts.  The personal account will be use to test as the customer; the business account will be receiving the subscription payments.
+
+Once you've added the account pair, navigate to the bussines account by selecting "View/edit account".  Now click on the "API Credentials" tab.  You'll need the "NVP/SOAP Sandbox API Credentials" for the next section.
+
+### Configuration
+
+#### Auth Server
+
+In order to enable and use PayPal in the auth server, set the following configuration options
+
+```
+{
+  "subscriptions": {
+    "paypalNvpSigCredentials": true,
+    "sandbox": true,
+    "user": "your PayPal NVP API User name",
+    "pwd": "your PayPal NVP API password",
+    "signature": "your PayPal NVP API signature"
+  }
+}
+```
+
+The environment variables equivalent would be
+```
+SUBSCRIPTIONS_PAYPAL_ENABLED=true \
+PAYPAL_SANDBOX=true \
+PAYPAL_NVP_USER='your PayPal NVP API User name' \
+PAYPAL_NVP_PWD='your PayPal NVP API password' \
+PAYPAL_NVP_SIGNATURE='your PayPal NVP API signature'
+```
+
+#### Payments Server
+
+The Payments frontend also does not offer PayPal as payment provider by default.  To enable the feature, set the following configuration options
+
+```
+{
+  "featureFlags": {
+    "usePaypalUIByDefault": true,
+    }
+  },
+  "paypal": {
+    "clientId": "sb",
+    "apiUrl": "https://www.sandbox.paypal.com",
+    "scriptUrl": "https://www.paypal.com"
+  }
+}
+```
+
+Or use the environment variables
+
+```
+FEATURE_USE_SCA_PAYMENT_UI_BY_DEFAULT=true \
+PAYPAL_CLIENT_ID='sb' \
+PAYPAL_API_URL='https://www.sandbox.paypal.com' \
+PAYPAL_SCRIPT_URL='https://www.paypal.com'
+```
+
+The paypal*/PAYPAL_* values are the defaults in the repo.  For local development, you do not need to change them.
+
+### Instant Payment Notification (IPN)
+
+[PayPal IPN](https://developer.paypal.com/docs/api-basics/notifications/ipn/) is PayPal's equivalent of Stripe's webhook feature.  We do rely on IPNs in the Subscription Platform.  Unlike Stripe, however, PayPal does not offer any tool that would forward the events to your local environment.  Our team use [ngrok](https://ngrok.com/) for that purpose.
+
+Once you have the services running locally, start ngrok with `ngrok 9000` and note the public URL.  Using your sandbox business account and the public URL fron ngrok, complete [these steps to set up IPNs](https://developer.paypal.com/docs/api-basics/notifications/ipn/IPNSetup/).
```

Comment:
Should we explain somewhere how to create personal and business sandbox accounts -- or link to a PayPal doc? https://developer.paypal.com/docs/api-basics/sandbox/accounts/

Reply:
Thanks for the link.  I'll use it in the PayPal Accounts section.